### ISSUE TITLE
How to use the trained model to get predictions on the test set?

### ISSUE THREAD

**Issue by Penguin0321** [davideferrari92/multiobjective_symbolic_regression#3]
I am here again 😂.
I've read the code about the class "SymbolicRegressor", but I didn't find a member function for predicting after fitting the training data. Did I miss something in the code?

**Comment by davideferrari92** [comment]
No worries, I'll update the notebooks for clarity. Anyway, because the regressor produces more than one model, you can see all of them in `sr.population`.

To get the first of them, use
`model = sr.population[0]  # Or model = sr.first_pareto_front[0] for the optimal models
`
To see its training performance, use
`model.fitness  # Or model.fitness_validation
`
To make the prediction, use
`model.evaluate(data)`

You can then iterate on all of them looking for the one with the characteristics that you want. Best performance of a particular metric, or a model that is simpler than others, etc.

**Comment by davideferrari92** [comment]
You can now pull the repository and will find the notebooks updated with some guidance on how to use the models.

**Comment by Penguin0321** [comment]
Thank you. I have got prediction on test set. Besides, is this project available for multi-target regression?

**Comment by davideferrari92** [comment]
Sure, but the success it always depends on whether the two outcomes are orthogonal or not.
You can use, for example, 
```python
fitness_functions = [
    WeightedMeanSquaredError(label='wmse1', target=target1,
                             weights=weights1, minimize=True,
                             constants_optimization=constants_optimization, 
                             constants_optimization_conf=constants_optimization_conf),
    WeightedMeanSquaredError(label='wmse2', target=target2,
                             weights=weights2, minimize=True,
                             constants_optimization=constants_optimization, 
                             constants_optimization_conf=constants_optimization_conf),
#  ...etc.
]
```

You see whether you want to pur the constants optimization in both on one only. You can then add as many objectives as you want and it will handle it automatically. The more, the harder is making fit all of them at the same time, and the larger will be the first pareto front.
Maybe consider using the Aiaike version of the WMSE to try to keep the complexity of the models at bay; you find it in the same module of the WMSE and receives the same arguments. You can add it with ```minimize=True``` and put ```minimize=False``` to the WMSE so you can see its value while the model trains on its AIC version which is more difficult to understand from the mere number.

Also, you can add all sorts of objectives with ```minimize=False``` to just measure them without them being used in the training. Doing so you can keep an eye on measures that you need but don't want to be at the core of the optimization.

**Comment by Penguin0321** [comment]
If I define the fitness_functions with multiple target, which model predicts for target1 and which for target2 after training?

**Comment by davideferrari92** [comment]
I see what you need. If you need to get two separate outcomes prediction, then you need to do two separate training and get two separate dedicated training. But this is true for any ML algorithm. Each target must have its own model.
